src/evaluate.cpp

```diff
@@ -32,807 +31,822 @@
+	// If scale is not already specific, scale down the endgame via general heuristics
+	if (sf == SCALE_FACTOR_NORMAL)
+	{
+			if (   pos.opposite_bishops()
+				&& pos.non_pawn_material() == 2 * BishopValueMg)
+				sf = 16 + 4 * pe->passed_count();
+			else
+				sf = std::min(sf, 36 + (pos.opposite_bishops() ? 2 : 7) * pos.count<PAWN>(strongSide));
 
-        sf = std::max(0, sf - (pos.rule50_count() - 12) / 4  );
-    }
+			sf = std::max(0, sf - pos.rule50_count()/4 +3  );
+	}
 
-    return ScaleFactor(sf);
-  }
+	return ScaleFactor(sf);
+	}
 
 
-  // Evaluation::value() is the main function of the class. It computes the various
-  // parts of the evaluation and returns the value of the position from the point
-  // of view of the side to move.
+	// Evaluation::value() is the main function of the class. It computes the various
+	// parts of the evaluation and returns the value of the position from the point
+	// of view of the side to move.
 
-  template<Tracing T>
-  Value Evaluation<T>::value() {
+	template<Tracing T>
+	Value Evaluation<T>::value() {
 
-    assert(!pos.checkers());
+	assert(!pos.checkers());
 
-    // Probe the material hash table
-    me = Material::probe(pos);
+	// Probe the material hash table
+	me = Material::probe(pos);
 
-    // If we have a specialized evaluation function for the current material
-    // configuration, call it and return.
-    if (me->specialized_eval_exists())
-        return me->evaluate(pos);
+	// If we have a specialized evaluation function for the current material
+	// configuration, call it and return.
+	if (me->specialized_eval_exists())
+			return me->evaluate(pos);
 
-    // Initialize score by reading the incrementally updated scores included in
-    // the position object (material + piece square tables) and the material
-    // imbalance. Score is computed internally from the white point of view.
-    Score score = pos.psq_score() + me->imbalance() + pos.this_thread()->contempt;
+	// Initialize score by reading the incrementally updated scores included in
+	// the position object (material + piece square tables) and the material
+	// imbalance. Score is computed internally from the white point of view.
+	Score score = pos.psq_score() + me->imbalance() + pos.this_thread()->contempt;
 
-    // Probe the pawn hash table
-    pe = Pawns::probe(pos);
-    score += pe->pawn_score(WHITE) - pe->pawn_score(BLACK);
+	// Probe the pawn hash table
+	pe = Pawns::probe(pos);
+	score += pe->pawn_score(WHITE) - pe->pawn_score(BLACK);
 
-    // Early exit if score is high
-    Value v = (mg_value(score) + eg_value(score)) / 2;
-    if (abs(v) > LazyThreshold + pos.non_pawn_material() / 64)
-       return pos.side_to_move() == WHITE ? v : -v;
+	// Early exit if score is high
+	Value v = (mg_value(score) + eg_value(score)) / 2;
+	if (abs(v) > LazyThreshold + pos.non_pawn_material() / 64)
+		return pos.side_to_move() ? -v : v;
```

Comment:
`pos.side_to_move() == WHITE` was intentional here, and also later in evaluate.cpp
We don't want to assume that WHITE is 0

Reply:
i'll rever it back

Reply:
as pointed out by adentong, this is almost impossible to review.
please start with latest master, and don't change the current style for spacing and line changes.

Reply:
I'm quite new git, so don't know how.... i'm closing this to reopen in new one ?!